### PR TITLE
[AppService] Fixes #14583: az webapp up should generate default name if name isn't provided

### LINKED ISSUE #14583: [az webapp up] if --name parameter is not provided we should auto generate one.
We should do similar logic to how static webapps auto generates the cname for a SWA

for example: zealous-tree-06c552e1e --> *adjective*-*noun*-shortguid

This is related to Python worklod review issue: [Issue 1457](https://azuretracking.visualstudio.com/Azure%20tracking/_workitems/edit/1457?src=WorkItemMention&src-action=artifact_link): [az webapp up] Make name optional, generate random name if left blank

### LINKED ISSUE #16185: WebApp:unexpected -l
On this page: https://docs.microsoft.com/en-us/cli/azure/webapp?view=azure-cli-latest#az_webapp_up

The example under the following section should not include "-l locationName":

"Create a web app with the default configuration, by running the command from the folder where the code to deployed exists."
`az webapp up -n MyUniqueAppName -l locationName`

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2983e943-cce1-effd-57d0-cce4b0c0a7e1
* Version Independent ID: 4262c40a-7140-b420-58b5-5cf40ab2316e
* Content: [az webapp](https://docs.microsoft.com/en-us/cli/azure/webapp?view=azure-cli-latest#az_webapp_up)
* Content Source: [latest/docs-ref-autogen/webapp.yml](https://github.com/MicrosoftDocs/azure-docs-cli/blob/master/latest/docs-ref-autogen/webapp.yml)
* Service: **app-service-web**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

### PR DESCRIPTION
**Pull request by calvinsID**
**Description**
- Make `--name` parameter optional for command `az webapp up`
- If name isn't provided, generate default name in the format adjective-noun-guid, ex. zealous-tree-06c552e1e

- Also fixes #16185 documentation update

**Testing Guide**
- run `az webapp up` without parameters in a directory you want to deploy
- a default name should be generated for you

image

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppService

**Comment by qwordy** [comment]
Credential found! If you confirm it is not a real credential, add the file name to scripts/ci/credscan/CredScanSuppressions.json

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_up_commands.py [hunk 1]
```diff
@@ -809,6 +810,7 @@ def test_webapp_up_change_runtime_version(self, resource_group):
         ])
 
         # test changing runtime to newer version
+        time.sleep(30)
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_up_commands.py:813]: Not very robust. Magic number.

**Review by qwordy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/_constants.py [hunk 2]
```diff
@@ -63,3 +63,74 @@ def __init__(self):
 
 
 RUNTIME_STACKS = os.path.abspath(os.path.join(os.path.abspath(__file__), '../resources/WebappRuntimeStacks.json'))
+
+APP_NAME_ADJECTIVES = [
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/_constants.py:67]: Nit: Can we add these to a JSON file as well & reference the file here?

Comment (panchagnula) [reply src/azure-cli/azure/cli/command_modules/appservice/_constants.py:67]: @calvinsID remove the test yaml file with the exposed secrets & re-record. https://github.com/Azure/azure-cli/pull/16267/files#diff-64d4540d613959d0ba1fb29e8591c6fa5612d2e3bb6bee1513d45949b4bc6672

**Comment by panchagnula** [comment]
Should we fix this issue as a part of this PR as well? https://github.com/Azure/azure-cli/issues/16266

**Review by panchagnula — CHANGES_REQUESTED** [review]
Please see comments for requested changes.

File: src/azure-cli/azure/cli/command_modules/appservice/_constants.py (hunk 2 above)
Comment (calvinsID) [reply src/azure-cli/azure/cli/command_modules/appservice/_constants.py]: Suppressed credential scanner for the webapp up test since its not a real secret

**Comment by calvinsID** [comment]
> Should we fix this issue as a part of this PR as well? #16266

I couldn't find a consistent repro, so will need to spend more time figuring why the issue happens periodically. I'll do it in a separate PR

**Review by panchagnula — APPROVED** [review]
LGTM

**Review by qwordy — APPROVED** [review]

**Pull request merged**